test/Common/TestOrchestrations.cs

```diff
@@ -28,6 +28,16 @@ public static async Task<string> SayHelloWithActivity([OrchestrationTrigger] Dur
             return output;
         }
 
+        public static async Task<bool> SayHelloWithActivityWithDeterministicGuid([OrchestrationTrigger] DurableOrchestrationContext ctx)
+        {
+            string input = ctx.GetInput<string>();
+            Guid firstGuid = ctx.NewGuid();
+            Guid secondGuid = ctx.NewGuid();
+            string output = await ctx.CallActivityAsync<string>(nameof(TestActivities.Hello), input);
+            Guid thirdGuid = ctx.NewGuid();
+            return firstGuid != secondGuid && firstGuid != thirdGuid && secondGuid != thirdGuid;

```

Comment:
This is a good basic test. Consider two more possible tests:
1. A test that ensures if you create 10K+ GUIDs in your orchestration they are all unique.
2. A test that ensures the same GUID values are generated during replays.

Reply:
Hello @cgillum,

Thank you for the great feedback.

I will add the tests.

Thank you!